From 66d7398d84108b36d9c6f1353ab728945831e5a2 Mon Sep 17 00:00:00 2001
From: zachary <zhangzg@marvell.com>
Date: Wed, 28 Dec 2016 22:44:10 +0800
Subject: [PATCH 1218/2241] fix: dts: a3700: set correct value for SPI clock
 and max frequency

- In SPI DT node, there are two clock settings:
  * clock-frequency, which is the SPI core clock, but this value
    is not used to configure SPI core clock, since it comes from TBG.
  * max-frequency, which is the max output clock to SPI devices.
    SPI controller divides core clock by prescale and generates it
    as output clock. SPI driver sets prescale value to:
        prescale = clock-frequency / max-frequency.
- In current dts file, these two clock values are both wrong:
  * clock-frequency is 16000 now, while the real clock comes
    from TBG should be 200MHz.
  * max-frequency is 4000 now, while the real output clock intended
    to get is 50MHz.
- Although the two clock values in DT node are wrong, but the real
  output clock is actually right, which is 50MHz. Since currently
  prescale value is 4 (16000 / 4000), and output clock is 50MHz
 (200MHz / 4)
- This patch updates these two clock values in DT node to correct ones.
  These two DT properties are to be obsoleted in late-on patches.

Change-Id: I7c0cb597dcb6ec8a55df8794834b738efef395a8
Signed-off-by: zachary <zhangzg@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35096
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Tested-by: Wilson Ding <dingwei@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 9895cab..3afb175 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -110,8 +110,8 @@
 				#address-cells = <1>;
 				#size-cells = <0>;
 				cell-index = <0>;
-				clock-frequency = <160000>;
-				max-frequency = <40000>;
+				clock-frequency = <200000000>;
+				max-frequency = <50000000>;
 				fifo-mode;
 				interrupts = <GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH>;
 				status = "disabled";
-- 
2.7.4

